addons/account/models/account_journal.py

```diff
@@ -71,13 +71,29 @@ def _default_invoice_reference_model(self):
     account_control_ids = fields.Many2many('account.account', 'journal_account_control_rel', 'journal_id', 'account_id', string='Allowed accounts',
         check_company=True,
         domain="[('deprecated', '=', False), ('company_id', '=', company_id), ('is_off_balance', '=', False)]")
+    # TODO: Transform this field in a computed Many2many in master and use that instead in the domain of default_account_id.
     default_account_type = fields.Many2one('account.account.type', compute="_compute_default_account_type")
     default_account_id = fields.Many2one(
         comodel_name='account.account', check_company=True, copy=False, ondelete='restrict',
         string='Default Account',
-        domain="[('deprecated', '=', False), ('company_id', '=', company_id),"
-               "'|', ('user_type_id', '=', default_account_type), ('user_type_id', 'in', type_control_ids),"
-               "('user_type_id.type', 'not in', ('receivable', 'payable'))]")
+        domain=lambda self: """[
+            ('deprecated', '=', False),
+            ('company_id', '=', company_id),
+            '|', ('user_type_id', 'in', %(bank)s if type == 'bank'
+                                        else %(cash)s if type == 'cash'
+                                        else %(sale)s if type == 'sale'
+                                        else %(purchase)s if type == 'purchase'
+                                        else ()),
+                 ('user_type_id', 'in', type_control_ids),
+            ('user_type_id.type', 'not in', ('receivable', 'payable'))
+        ]""" % {
+            'bank': (self.env.ref('account.data_account_type_liquidity').id,
+                     self.env.ref('account.data_account_type_credit_card').id),
+            'cash': (self.env.ref('account.data_account_type_liquidity').id,),
+            'sale': (self.env.ref('account.data_account_type_revenue').id,),
+            'purchase': (self.env.ref('account.data_account_type_expenses').id,),
```

Comment:
What about a sub-method? :sweat_smile:

Reply:
Also, you need to adapt the `_seek_for_lines` method on `account.bank.statement.line` accordingly and probably some others places... :grimacing:

Reply:
> What about a sub-method? 😅

Not a bad idea 😅 I extracted it in a method.

> Also, you need to adapt the `_seek_for_lines` method on `account.bank.statement.line` accordingly and probably some others places... 😬

Not sure what needs to be changed there ... Can you enlighten me?

Reply:
@dylankiss When you create a statement line on a bank journal, the liquidity account is computed from the journal default_account_id. Later on, when you need to split the statement line's amls to distinguish which one is the liquidity and which ones are the write off or an open balance. With your change, the statement line will raise an error at each write on the statement line because the code is no longer able to sync the statement line with the accounting journal entry

Reply:
@smetl Sorry, but I still don't see how extending the domain of `default_account_id` could cause problems with this.

Reply:
@dylankiss Put a credit card account on your bank journal. Then, create a statement line and try to reconcile it on the bank reco widget